qutebrowser/browser/commands.py

```diff
@@ -832,6 +833,62 @@ def paste(self, sel=False, tab=False, bg=False, window=False):
             self._open(url, tab, bg, window)
 
     @cmdutils.register(instance='command-dispatcher', scope='window',
+                       completion=[usertypes.Completion.tab])
+    def buffer(self, index=None):
+        """Select tab by index or url/title best match, focus window if
+        necessary.
+
+        Args:
+            index: The win_id/index of the tab to focus. Or a substring
+                   in which case the closest match will be focused.
+        """
+        if not index or not len(index):
+            return
+
+        try:
+            # Is index of form 12/34 ?
+            list(map(int, index.split('/')))
+        except ValueError:
+            model = instances.get(usertypes.Completion.tab)
+            sf = sortfilter.CompletionFilterModel(source=model)
+            sf.set_pattern(index)
+            if sf.count() > 0:
+                index = sf.data(sf.first_item())
+
+        if len(index.split('/')):
+            try:
+                # Importing at the top of the file causes a loop
+                from qutebrowser import app
+                if len(index.split('/')) == 2:
+                    w, i = map(int, index.split('/'))
+                elif len(index.split('/')) == 1:
+                    i = int(index)
+                    w = 0
+                    aw = app.qApp.activeWindow()
+                    for win_id in objreg.window_registry:
+                        if aw == objreg.window_registry[win_id]:
+                            w = win_id
```

Comment:
You should be able to simply do `active_win = objreg.get('app').activeWindow()` and `w = active_win.win_id` instead. Note you'll need to handle `active_win` being `None`.